samples/subsys/ipc/rpmsg_multi_instance/remote/src/main.c

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <drivers/ipm.h>
+#include <sys/printk.h>
+#include <device.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <init.h>
+
+#include <ipc/rpmsg_multi_instance.h>
+
+#define IPM_WORK_QUEUE_STACK_SIZE 1024
+K_THREAD_STACK_DEFINE(ipm_stack_area_1, IPM_WORK_QUEUE_STACK_SIZE);
+K_THREAD_STACK_DEFINE(ipm_stack_area_2, IPM_WORK_QUEUE_STACK_SIZE);
+
+#define APP_TASK_STACK_SIZE (1024)
+K_THREAD_STACK_DEFINE(thread_stack_1, APP_TASK_STACK_SIZE);
+K_THREAD_STACK_DEFINE(thread_stack_2, APP_TASK_STACK_SIZE);
+static struct k_thread thread_data_1, thread_data_2;
+
+K_SEM_DEFINE(bound_ept1_sem, 0, 1);
+K_SEM_DEFINE(bound_ept2_sem, 0, 1);
+
+K_SEM_DEFINE(data_rx1_sem, 0, 1);
+K_SEM_DEFINE(data_rx2_sem, 0, 1);
+
+unsigned int received_data_1, received_data_2;
+
+static struct rpmsg_mi_ctx ctx_1, ctx_2;
+struct rpmsg_mi_ept ept_1, ept_2;
+
+
+int boud_cb(void *priv)
+{
+	struct rpmsg_mi_ept *ept = (struct rpmsg_mi_ept *)priv;
+
+	if (ept == &ept_1) {
+		k_sem_give(&bound_ept1_sem);
+	} else if (ept == &ept_2) {
+		k_sem_give(&bound_ept2_sem);
+	}
+	return 0;
+}
+
+int received_cb(void *data, size_t len, void *priv)
+{
+	struct rpmsg_mi_ept *ept = (struct rpmsg_mi_ept *)priv;
+
+	if (ept == &ept_1) {
+		received_data_1 = *((unsigned int *)data);
+		k_sem_give(&data_rx1_sem);
+	} else if (ept == &ept_2) {
+		received_data_2 = *((unsigned int *)data);
+		k_sem_give(&data_rx2_sem);
+	}
+	return 0;
+}
+
+static const struct rpsmg_mi_ctx_cfg cfg_1 = {
+	.name               = "instance 1",
+	.ipm_stack_area     = ipm_stack_area_1,
+	.ipm_stack_size     = K_THREAD_STACK_SIZEOF(ipm_stack_area_1),
+	.ipm_thread_name    = "ipm_work_q_1",
+	.ipm_work_q_prio    = 0,
+	.ipm_tx_name        = "IPM_0",
+	.ipm_rx_name        = "IPM_1",
+};
+
+static const struct rpsmg_mi_ctx_cfg cfg_2 = {
+	.name               = "instance 2",
+	.ipm_stack_area     = ipm_stack_area_2,
+	.ipm_stack_size     = K_THREAD_STACK_SIZEOF(ipm_stack_area_2),
+	.ipm_thread_name    = "ipm_work_q_2",
+	.ipm_work_q_prio    = 0,
+	.ipm_tx_name        = "IPM_2",
+	.ipm_rx_name        = "IPM_3",
+};
+
+static struct rpmsg_mi_cb cb = {
+	.bound   = boud_cb,
+	.received = received_cb,
+};
+
+static struct rpmsg_mi_ept_cfg ept_cfg_1 = {
+	.ept_no = 0,
+	.cb     = &cb,
+	.priv   = &ept_1,
+};
+
+static struct rpmsg_mi_ept_cfg ept_cfg_2 = {
+	.ept_no = 0,
+	.priv   = &ept_2,
+	.cb     = &cb,
```

Comment:
```suggestion
	.cb     = &cb,
	.priv   = &ept_2,
```
to keep the same order as in the previous structure